lib/internal/modules/esm/loader.js

```diff
@@ -118,7 +118,16 @@ class Loader {
     source,
     url = pathToFileURL(`${process.cwd()}/[eval${++this.evalIndex}]`).href
   ) {
-    const evalInstance = (url) => new ModuleWrap(url, undefined, source, 0, 0);
+    const evalInstance = (url) => {
+      const module = new moduleWrap.ModuleWrap(url, undefined, source, 0, 0);
+      moduleWrap.callbackMap.set(module, {
+        importModuleDynamically: (specifier, { url }) => {
+          return this.import(specifier, url);
+        }
+      });
+
+      return module;
```

Comment:
Rewrite in [this way](https://github.com/nodejs/node/blob/master/lib/internal/modules/esm/create_dynamic_module.js#L35) for consistence?

Reply:
Done